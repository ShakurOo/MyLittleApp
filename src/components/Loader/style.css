.wrapper {
  bottom: 0;
  display: inline-block;
  text-align: center;
  top: 50%;
  height: 100%;
  position: absolute;
  width: 100%;

  img {
    height: 64px;
    width: 64px;
    transform: translateY(-50%);
  }
}
