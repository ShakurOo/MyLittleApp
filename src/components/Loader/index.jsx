import React from 'react'
import LoaderIMG from './assets/loader.gif'
import style from './style'

const Loader = () => (
  <div className={style.wrapper}>
    <img src={LoaderIMG} alt='Loader' />
  </div>
)

export default Loader
