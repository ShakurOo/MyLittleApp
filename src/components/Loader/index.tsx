import React from 'react'
import LoaderIcon from './assets/loader.gif'
import { Wrapper } from './style'

const Loader = (): JSX.Element => (
  <Wrapper>
    { LoaderIcon && <img src={LoaderIcon} alt='Loader' /> }
  </Wrapper>
)

export default Loader
