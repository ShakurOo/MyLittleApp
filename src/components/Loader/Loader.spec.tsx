import { shallow } from 'enzyme'
import React from 'react'
import Loader from './'

describe('Components', () => {
  describe('Loader', () => {
    it('renders <Loader> as a snapshot', () => {
      const shallowLoader = shallow(
        <Loader />
      )
      expect(shallowLoader).toMatchSnapshot()
    })

    it('renders the loader gif', () => {
      const shallowLoader = shallow(
        <Loader />
      )
      expect(shallowLoader.find('img'))
        .toBeDefined()
    })
  })
})
