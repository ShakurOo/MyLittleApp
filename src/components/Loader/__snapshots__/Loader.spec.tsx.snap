// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Components Loader renders <Loader> as a snapshot 1`] = `ShallowWrapper {}`;
